crates/store/src/db/sql.rs

```diff
@@ -1093,21 +991,16 @@ pub fn get_note_sync(
     block_num: BlockNumber,
     note_tags: &[u32],
 ) -> Result<NoteSyncUpdate, NoteSyncError> {
-    let notes = select_notes_since_block_by_tag(conn, note_tags, block_num)?;
-
-    let (block_header, chain_tip) = if !notes.is_empty() {
-        let block_header = select_block_header_by_block_num(conn, Some(notes[0].block_num))?
-            .ok_or(NoteSyncError::EmptyBlockHeadersTable)?;
-        let tip = select_block_header_by_block_num(conn, None)?
-            .ok_or(NoteSyncError::EmptyBlockHeadersTable)?;
-
-        (block_header, tip.block_num())
-    } else {
-        let block_header = select_block_header_by_block_num(conn, None)?
-            .ok_or(NoteSyncError::EmptyBlockHeadersTable)?;
-
-        let block_num = block_header.block_num();
-        (block_header, block_num)
+    let notes = select_notes_since_block_by_tag_and_sender(conn, note_tags, &[], block_num)?;
+
+    let tip = select_block_header_by_block_num(conn, None)?
+        .ok_or(NoteSyncError::EmptyBlockHeadersTable)?;
```

Comment:
Since we know the latest block number in the `state` - we may be able to pass `chain_tip_num` into this function and potentially avoid this database query (i.e., we can always do just 1 query here instead of sometimes doing 2 queries and sometimes 1).

Reply:
We don't store latest block number in the `State` in-memory by now or I can't find a place you mean. Should I update our in-memory state in order to save the current block number?

Reply:
We do have [latest_block_num()](https://github.com/0xPolygonMiden/miden-node/blob/main/crates/store/src/state.rs#L707) function on the `State`. Let's create an issue for this optimization (could be a "good first issue").

Reply:
@bobbinth ah, thank you, I forgot about this feature of chain mmr. :)

Reply:
@bobbinth done: https://github.com/0xPolygonMiden/miden-node/pull/477